### ISSUE TITLE
UnitOfWork

### ISSUE THREAD

**Issue by gertjvr** [rebus-org/Rebus#406]
Found an issue with UnitOfWork using the [EnableUnitOfWork](https://github.com/rebus-org/Rebus/blob/master/Rebus.UnitOfWork/UnitOfWorkConfigurationExtensions.cs#L35) with `Task` parameters, if you don't provide a `Rollback` or `CleanUp` action it defaults to null and throws exceptions when the `try block` [finalizes](https://github.com/rebus-org/Rebus/blob/master/Rebus.UnitOfWork/UnitOfWorkStep.cs#L46)  or an [exception](https://github.com/rebus-org/Rebus/blob/master/Rebus.UnitOfWork/UnitOfWorkStep.cs#L41) occurs.

```csharp
try
{
    await next();
    await _commitAction(messageContext, unitOfWork);
}
catch (Exception)
{
    await _rollbackAction(messageContext, unitOfWork);
    throw;
}
finally
{
    await _cleanupAction(messageContext, unitOfWork);
}
```

Maybe do a null check before or have some noop function set as defaults.

**Comment by mookid8000** [comment]
good catch, thanks!

**Comment by mookid8000** [comment]
it's fixed in 0.99.31 which is on Nuget.org now
